### PR TITLE
tests: faster integration tests, and more complete e2e tests

### PR DESCRIPTION
**Pull request by nicolasburtey**
- reduce the number of dependancies for integrations tests. some tests that rely on external dependancy (kratos, oathkeeper) has been moved to e2e. 
- e2e now users now login with the graphql API, instead of some internal functions used in integration tests

### CODE REVIEW

File: test/e2e/servers/graphql-main-server/with-auth-requests.spec.ts [hunk 1]
```diff
@@ -53,23 +53,33 @@ import {
...
+
 const satsAmount = toSats(50_000)
 const centsAmount = toCents(4_000)
 
 beforeAll(async () => {
-  await publishOkexPrice()
   await initializeTestingState(defaultStateConfig())
+  serverPid = await startServer("start-main-ci")
+  await loginFromPhoneAndCode({ phone, code })
+  await loginFromPhoneAndCode({ phone: phoneOther, code: codeOther })
```
Comment (dolcalmi) [inline test/e2e/servers/graphql-main-server/with-auth-requests.spec.ts:78]: you are not using generated apollo client. why is the purpose of this?

File: test/e2e/servers/graphql-main-server/no-auth-requests.spec.ts [hunk 2]
```diff
@@ -38,8 +39,10 @@ const { phone, code } = getPhoneAndCodeFromRef("G")
 
 beforeAll(async () => {
   await initializeTestingState(defaultStateConfig())
-  correctCode = `${code}` as PhoneCode
   serverPid = await startServer("start-main-ci")
+  await loginFromPhoneAndCode({ phone, code })
```
Comment (dolcalmi) [inline test/e2e/servers/graphql-main-server/no-auth-requests.spec.ts:43]: you are not using client, what is the purpose of this?

File: test/e2e/servers/graphql-main-server/galoy-pay-testing.spec.ts [hunk 3]
```diff
@@ -30,24 +30,32 @@ import {
...
 const { phone, code } = getPhoneAndCodeFromRef(sendingUserRef)
+const { phone: phoneRecipient, code: codeRecipient } =
+  getPhoneAndCodeFromRef(receivingUserRef)
 
 beforeAll(async () => {
   await initializeTestingState(defaultStateConfig())
+
+  serverPid = await startServer("start-main-ci")
+
+  await loginFromPhoneAndCode({ phone, code })
```
Comment (dolcalmi) [inline test/e2e/servers/graphql-main-server/galoy-pay-testing.spec.ts:52]: what is the purpose of this? you are not using the client

File: test/e2e/servers/graphql-main-server/galoy-pay-testing.spec.ts [hunk 4]
```diff
@@ -30,24 +30,32 @@ import {
...
+  const c = await loginFromPhoneAndCode({ phone: phoneRecipient, code: codeRecipient })
```
Comment (dolcalmi) [inline test/e2e/servers/graphql-main-server/galoy-pay-testing.spec.ts:53]: please rename this variable, `c` is not a proper name

File: docker-compose.yml [hunk 5]
```diff
@@ -102,11 +97,11 @@ services:
       - POSTGRES_PASSWORD=galoy-price-pwd
       - POSTGRES_DB=galoy-price-history
   bitcoind:
-    image: lncm/bitcoind:v23.0
+    image: lncm/bitcoind:v24.0
```
Comment (dolcalmi) [inline docker-compose.yml:100]: maybe is better to do the changes (versions) in this file in a different PR

File: Makefile [hunk 6]
```diff
@@ -105,18 +105,13 @@ delete-e2e:
 main-in-ci-build:
 	yarn build && \
 	make create-tmp-env-ci && \
-	TMP_ENV_CI=tmp.env.ci docker compose -f docker-compose.yml run --name e2e-tests e2e-tests make start-main-ci || \
-	make del-containers
+	TMP_ENV_CI=tmp.env.ci docker compose -f docker-compose.yml -f docker-compose.override.yml run --name e2e-tests e2e-tests make start-main-ci || \
```
Comment (dolcalmi) [inline Makefile:108]: did you test this in our CI ?, override is only used locally so if you do this change and works correctly then we dont need an override file

**Review by dolcalmi — CHANGES_REQUESTED** [review]
looks good, just minor changes/questions

File: test/e2e/servers/graphql-main-server/no-auth-requests.spec.ts (hunk 2 above)
Comment (nicolasburtey) [reply test/e2e/servers/graphql-main-server/no-auth-requests.spec.ts:43]: to create the user/account in the db

File: test/e2e/servers/graphql-main-server/galoy-pay-testing.spec.ts (hunk 3 above)
Comment (nicolasburtey) [reply test/e2e/servers/graphql-main-server/galoy-pay-testing.spec.ts:52]: to create the user/account in the db

File: Makefile (hunk 6 above)
Comment (nicolasburtey) [reply Makefile:108]: it's not used in CI. only in local dev.

File: docker-compose.yml (hunk 5 above)
Comment (nicolasburtey) [reply docker-compose.yml:100]: reverted (but thanksfully it's really a no-op)

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**